tests/Unit/DataStructures/TensorEagerMath/Test_DeterminantAndInverse.cpp

```diff
@@ -0,0 +1,248 @@
+  CHECK((det_inv.second.template get<0, 2>()) == 0.5625);
+  CHECK((det_inv.second.template get<1, 0>()) == -0.75);
+  CHECK((det_inv.second.template get<1, 1>()) == 1.0);
+  CHECK((det_inv.second.template get<1, 2>()) == -0.25);
+  CHECK((det_inv.second.template get<2, 0>()) == 0.5625);
+  CHECK((det_inv.second.template get<2, 1>()) == -0.25);
+  CHECK((det_inv.second.template get<2, 2>()) == -0.0625);
+}
+
+template <typename TensorType>
+void verify_det_and_inv_symmetric_4d() {
+  TensorType t{};
+  t.template get<0, 0>() = 2.0;
+  t.template get<0, 1>() = 3.0;
+  t.template get<0, 2>() = 6.0;
+  t.template get<0, 3>() = 11.0;
+  t.template get<1, 1>() = 5.0;
+  t.template get<1, 2>() = 7.0;
+  t.template get<1, 3>() = 12.0;
+  t.template get<2, 2>() = 10.0;
+  t.template get<2, 3>() = 13.0;
+  t.template get<3, 3>() = 17.0;
+  const auto det_inv = determinant_and_inverse(t);
+  CHECK(det_inv.first.get() == -100.0);
+  CHECK((det_inv.second.template get<0, 0>()) == approx(0.84));
+  CHECK((det_inv.second.template get<0, 1>()) == approx(-0.94));
+  CHECK((det_inv.second.template get<0, 2>()) == approx(-0.34));
+  CHECK((det_inv.second.template get<0, 3>()) == approx(0.38));
+  CHECK((det_inv.second.template get<1, 0>()) == approx(-0.94));
+  CHECK((det_inv.second.template get<1, 1>()) == approx(1.04));
+  CHECK((det_inv.second.template get<1, 2>()) == approx(-0.06));
+  CHECK((det_inv.second.template get<1, 3>()) == approx(-0.08));
+  CHECK((det_inv.second.template get<2, 0>()) == approx(-0.34));
+  CHECK((det_inv.second.template get<2, 1>()) == approx(-0.06));
+  CHECK((det_inv.second.template get<2, 2>()) == approx(0.84));
+  CHECK((det_inv.second.template get<2, 3>()) == approx(-0.38));
+  CHECK((det_inv.second.template get<3, 0>()) == approx(0.38));
+  CHECK((det_inv.second.template get<3, 1>()) == approx(-0.08));
+  CHECK((det_inv.second.template get<3, 2>()) == approx(-0.38));
+  CHECK((det_inv.second.template get<3, 3>()) == approx(0.16));
+}
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.DataStructures.Tensor.EagerMath.DeterminantAndInverse",
+                  "[DataStructures][Unit]") {
+  // Check that the inverse tensor has the expected index structure (indices
+  // should be swapped and raised/lowered)...
+  {
+    using generic = tmpl::integral_list<int32_t, 2, 1>;
+    using iA = SpatialIndex<2, UpLo::Up, Frame::Grid>;
+    using ia = SpatialIndex<2, UpLo::Lo, Frame::Grid>;
+    using iB = SpatialIndex<2, UpLo::Up, Frame::Inertial>;
+    using ib = SpatialIndex<2, UpLo::Lo, Frame::Inertial>;
+    using function_t =
+        decltype(&determinant_and_inverse<double, generic, iA, ib>);
+    using input_t = Tensor<double, generic, index_list<iA, ib>>;
+    using result_t = std::result_of_t<function_t(input_t)>;
+    using inv_t = Tensor<double, generic, index_list<iB, ia>>;
+    static_assert(cpp17::is_same_v<inv_t, result_t::second_type>,
+                  "Inverse tensor has incorrect index structure.");
```

Comment:
Can you try this beauty out:

```cpp
static_assert(
    cpp17::is_same_v<
        Tensor<double, tmpl::integral_list<int32_t, 2, 1>,
               index_list<SpatialIndex<2, UpLo::Up, Frame::Inertial>,
                          SpatialIndex<2, UpLo::Lo, Frame::Grid>>>,
        decltype(determinant_and_inverse(
            std::declval<Tensor<
                double, tmpl::integral_list<int32_t, 2, 1>,
                index_list<SpatialIndex<2, UpLo::Up, Frame::Grid>,
                           SpatialIndex<2, UpLo::Lo, Frame::Inertial>>>>()))>,
    "Inverse tensor has incorrect index structure.");
```

Reply:
This _is_ the droid I was looking for (modulo `::second_type`).